server/src/test/java/org/opensearch/common/blobstore/stream/read/listener/FileCompletionListenerTests.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.stream.read.listener;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class FileCompletionListenerTests extends OpenSearchTestCase {
+
+    public void testFileCompletionListener() {
+        int numStreams = 10;
+        String fileName = "test_segment_file";
+        ActionListener<String> completionListener = mock(ListenerTestUtils.TestCompletionListener.class);
+        FileCompletionListener fileCompletionListener = new FileCompletionListener(numStreams, fileName, completionListener);
+
+        for (int stream = 0; stream < numStreams; stream++) {
+            // Ensure completion listener called only when all streams are completed
+            verify(completionListener, times(0)).onResponse(anyString());
+            fileCompletionListener.onResponse(stream);
+        }
+
+        verify(completionListener, times(1)).onResponse(eq(fileName));
+    }
+
+    public void testFileCompletionListenerFailure() {
+        int numStreams = 10;
+        String fileName = "test_segment_file";
+        ActionListener<String> completionListener = mock(ListenerTestUtils.TestCompletionListener.class);
```

Comment:
You shouldn't mock a test class. You can make it do whatever you need, e.g. add an invocation counter to the test class and assert on that value.